src/misclibs.F

```diff
@@ -3523,20 +3524,29 @@ subroutine calcksmax(nstat,rstat,dt,uh,vh,mf,kmh,kmv,khh,khv)
             do i=1,ni
                tem = khh(i,j,k)*dtdx*uh(i)*uh(i)
                if( tem.eq.ksh(k) )then
+                 !$acc atomic write
```

Comment:
I am still confused about the behavior of `acc atomic write`. I would suggest using `loop seq` for the nested three loops to behave the same as the serial code first.

Reply:
@sjsprecious , I'm going to leave those lines in for the moment because as @johnmauff points out, this might not be an issue because the NVIDIA version does not use these.  Once we've verified the correctness of the new version, we can remove this code.